tests/phpunit/tests/media.php

```diff
@@ -3798,6 +3798,83 @@ function() use ( &$result ) {
 
 		$this->assertFalse( $result );
 	}
+
+	/**
+	 * @ticket 55443
+	 */
+	public function test_wp_print_image_mime_fallback_script_if_content_has_updated_images() {
+
+		if ( ! wp_image_editor_supports( array( 'mime_type' => 'image/webp' ) ) ) {
+			$this->markTestSkipped( 'This test requires WebP support.' );
+		}
+
+		remove_all_actions( 'template_redirect' );
+		do_action( 'template_redirect' );
+
+		$filename      = DIR_TESTDATA . '/images/' . self::$large_filename;
+		$attachment_id = $this->factory->attachment->create_upload_object( $filename );
+
+		// Preferring WebP over JPEG results in no changes to content.
+		add_filter(
+			'wp_content_image_mimes',
+			function() {
+				return array( 'image/webp', 'image/jpeg' );
+			}
+		);
+
+		apply_filters(
+			'the_content',
+			sprintf(
+				'<p>before image</p>%s<p>after image</p>',
+				wp_filter_content_tags( wp_get_attachment_image( $attachment_id, 'medium', false, array( 'class' => "wp-image-{$attachment_id}" ) ) )
+			)
+		);
+
+		// `wp_print_image_mime_fallback_script()` assumes `wp-includes/js/wp-image-mime-fallback-loader.js` is present:
+		copy(
+			ABSPATH . 'js/_enqueues/wp/image-mime-fallback-loader.js',
+			ABSPATH . WPINC . '/js/wp-image-mime-fallback-loader.js'
+		);
+
+		$this->assertStringContainsString( 'data:image/webp;base64,UklGR', get_echo( 'wp_footer' ) );
+	}
+
+	/**
+	 * @ticket 55443
+	 */
+	public function test_wp_print_image_mime_fallback_script_if_content_has_no_updated_images() {
+
+		if ( ! wp_image_editor_supports( array( 'mime_type' => 'image/webp' ) ) ) {
+			$this->markTestSkipped( 'This test requires WebP support.' );
+		}
+
+		$GLOBALS['_wp_image_mime_fallback_should_load'] = false;
+		apply_filters( 'the_content', '<p>no image</p>' );
+
+		$this->assertStringNotContainsString( 'data:image/webp;base64,UklGR', get_echo( 'wp_footer' ) );
+	}
+
+	/**
+	 * @ticket 55443
+	 */
+	public function test_wp_print_image_mime_fallback_script_empty() {
+		$GLOBALS['_wp_image_mime_fallback_should_load'] = false;
+		$this->assertSame( '', get_echo( 'wp_print_image_mime_fallback_script' ) );
+	}
+
+	/**
+	 * @ticket 55443
+	 */
+	public function test_wp_print_image_mime_fallback_script_return_fallback_script() {
+		$GLOBALS['_wp_image_mime_fallback_should_load'] = true;
+
+		// `wp_print_image_mime_fallback_script()` assumes `wp-includes/js/wp-image-mime-fallback-loader.js` is present:
+		self::touch( ABSPATH . WPINC . '/js/wp-image-mime-fallback-loader.js' );
+
+		$output = get_echo( 'wp_print_image_mime_fallback_script' );
+		$this->assertNotSame( '', $output );
+		$this->assertStringContainsString( 'window._wpImageMimeFallbackSettings', $output );
```

Comment:
Test methods with multiple assertions need a `message` parameter for each assertion, per [Core Handbook - Writing PHPUnit Tests - Using Assertions](https://make.wordpress.org/core/handbook/testing/writing-phpunit-tests/#using-assertions).